mlir/test/Integration/Dialect/Arith/CPU/comparison.mlir

```diff
@@ -0,0 +1,159 @@
+// RUN:             --convert-func-to-llvm --convert-arith-to-llvm | \
+// RUN:   mlir-cpu-runner -e entry -entry-point-result=void \
+// RUN:                   --shared-libs=%mlir_c_runner_utils | \
+// RUN:   FileCheck %s --match-full-lines
+
+func.func @cmpi_eq_i1(%v1 : i1, %v2 : i1) {
+  vector.print str "@cmpi_eq_i1\n"
+  %res = arith.cmpi eq, %v1, %v2 : i1
+  vector.print %res : i1
+  return
+}
+
+func.func @cmpi_slt_i1(%v1 : i1, %v2 : i1) {
+  vector.print str "@cmpi_slt_i1\n"
+  %res = arith.cmpi slt, %v1, %v2 : i1
+  vector.print %res : i1
+  return
+}
+
+func.func @cmpi_sle_i1(%v1 : i1, %v2 : i1) {
+  vector.print str "@cmpi_sle_i1\n"
+  %res = arith.cmpi sle, %v1, %v2 : i1
+  vector.print %res : i1
+  return
+}
+
+func.func @cmpi_sgt_i1(%v1 : i1, %v2 : i1) {
+  vector.print str "@cmpi_sgt_i1\n"
+  %res = arith.cmpi sgt, %v1, %v2 : i1
+  vector.print %res : i1
+  return
+}
+
+func.func @cmpi_sge_i1(%v1 : i1, %v2 : i1) {
+  vector.print str "@cmpi_sge_i1\n"
+  %res = arith.cmpi sge, %v1, %v2 : i1
+  vector.print %res : i1
+  return
+}
+
+func.func @cmpi_eq() {
+  // ------------------------------------------------
+  // Test i1
+  // ------------------------------------------------
+  %false_i1 = arith.constant 0 : i1
+  %true_i1 = arith.constant 1 : i1
+  %true_i1_n1 = arith.constant -1 : i1
+
+  // int values 1 and -1 are represented with the same bitvector (`0b1`)
+  // CHECK-LABEL: @cmpi_eq_i1
+  // CHECK-NEXT:  1
+  func.call @cmpi_eq_i1(%true_i1, %true_i1_n1) : (i1, i1) -> ()
+
+  // CHECK-LABEL: @cmpi_eq_i1
+  // CHECK-NEXT:  0
+  func.call @cmpi_eq_i1(%false_i1, %true_i1) : (i1, i1) -> ()
+
+  // CHECK-LABEL: @cmpi_eq_i1
+  // CHECK-NEXT:  0
+  func.call @cmpi_eq_i1(%true_i1, %false_i1) : (i1, i1) -> ()
+
+  // CHECK-LABEL: @cmpi_eq_i1
+  // CHECK-NEXT:  1
+  func.call @cmpi_eq_i1(%true_i1, %true_i1) : (i1, i1) -> ()
+
+  // CHECK-LABEL: @cmpi_eq_i1
+  // CHECK-NEXT:  1
+  func.call @cmpi_eq_i1(%false_i1, %false_i1) : (i1, i1) -> ()
+
+  // ------------------------------------------------
+  // TODO: Test i8, i16 etc..
+  // ------------------------------------------------
+  return
+}
+
+func.func @cmpi_signed() {
+  // ------------------------------------------------
+  // Test i1
+  // ------------------------------------------------
+  %false_i1 = arith.constant 0 : i1
+  %true_i1 = arith.constant 1 : i1
+  %true_i1_n1 = arith.constant -1 : i1
+
+  // int values 1 and -1 are represented with the same bitvector (`0b1`)
+  // But, bitvector `1` is interpreted as int value -1 in signed comparison
+
+  // CHECK-LABEL: @cmpi_sge_i1
+  // CHECK-NEXT:  1
+  func.call @cmpi_sge_i1(%false_i1, %true_i1_n1) : (i1, i1) -> ()
+
+  // CHECK-LABEL: @cmpi_sge_i1
+  // CHECK-NEXT:  1
+  func.call @cmpi_sge_i1(%false_i1, %true_i1) : (i1, i1) -> ()
+  
+  // CHECK-LABEL: @cmpi_sge_i1
+  // CHECK-NEXT:  0
+  func.call @cmpi_sge_i1(%true_i1, %false_i1) : (i1, i1) -> ()
+
+  %false = arith.constant false
+  %true = arith.constant true
```

Comment:
Any idea where and how these get parsed? (what's the underlying bit pattern?) Since you've already added these, why not add "eq" comparison with the previous definitions? (eg `%false_i1`)